eclair-core/src/main/scala/fr/acinq/eclair/payment/Relayer.scala

```diff
@@ -138,8 +136,9 @@ class Relayer(nodeParams: NodeParams, register: ActorRef, paymentHandler: ActorR
         case Local(id, None) =>
           // we sent the payment, but we probably restarted and the reference to the original sender was lost,
           // we publish the failure on the event stream and update the status in paymentDb
-          nodeParams.db.payments.updateOutgoingPayment(id, OutgoingPaymentStatus.FAILED)
-          context.system.eventStream.publish(PaymentFailed(id, paymentHash, Nil))
+          val result = PaymentFailed(id, paymentHash, Nil)
+          nodeParams.db.payments.updateOutgoingPayment(result)
```

Comment:
I'll investigate that.

Reply:
I'm not a big fan, it feels hackish to use a deadletters sender.
I'll re-visit the relayer part in the MPP PR because we'll have to add more logic here, so let's keep it like that for now?